#### ***Задача 1***. Когда используется критерий Стьюдента, а когда Z –критерий?

### Решение

Z-критерий используется если:
* известна σ генеральная совокупность
* нормальность

t-критерий(критерий Стьюдента) используется:
* σ генеральная СОВОКУПНОСТЬ неизвестна
* нормальность
* равенство дисперсий

При больших размерах выборок результаты z-теста и k-теста дают схожие значения p-value

#### ***Задача 2.*** Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

### Решение

- Сформируем гипотезу: 
- H0: μ=μ0
- H1: μ>μ0

- Выбор критерия: Zн=Xср-μ/σ/√n

- Уровень статистической значимости: α=0.05

tкрит=1.65(значение по таблице z-значений)

Zн=Xср-μ/σ/√n=17.5-17/2/√100=2.5

tкрит<tнабл, следовательно H0 отвергается
- Вывод: шарики для подшипников имеют диаметр больше 17 мм. Гипотеза верна при α=0.05

#### ***Задача 3.*** 
Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья
составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна
99%? (Провести двусторонний тест.)

### Решение

In [44]:
import numpy as np
import scipy.stats as stats

Сформируем гипотезу:

- H0: μ = 200 гр -> утверждение, что средний вес пачки печенья составляет 200 г верно
- H1: μ < 200 гр -> утверждение, что средний вес пачки печенья составляет 200 г неверно

In [45]:
x_box = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
μ = 200
number = 10
x_var = np.var(x_box, ddof=1)
x_sigma = x_var**0.5
x_mean_box = np.mean(x_box)
print(f'x_var={x_var}')
print(f'x_sigma={x_sigma}')
print(f'x_mean_box={x_mean_box}')

x_var=19.833333333333332
x_sigma=4.453463071962462
x_mean_box=198.5


Используем t-критерий:

In [46]:
tнабл = (x_mean_box - μ)/x_sigma/np.sqrt(number)
tнабл

-0.10651074037450894

In [47]:
alpha = 0.01
t1_box = stats.t.ppf(alpha/2, df=2*(number-1))
t2_box = stats.t.ppf(1-alpha/2, df=2*(number-1))
t1_box, t2_box

(-2.8784404727135864, 2.878440472713585)

- tкрит = 2.878
- tнабл = -0.106
- tкрит>tнабл
- H0>H1. Нулевая гипотеза не отвергается. 
- Вывод: гипотеза о том, что средний вес пачки составляет 200 гр. верна

### Задача 4. 
Есть ли статистически значимые различия в росте
дочерей?
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165.
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160, 163

### Решение

In [48]:
import numpy as np
import scipy.stats as stats

Сформируем гипотезу:
- H0: μx = μy -> статистически значимых различий в росте дочерей нет
- H1: μx != μy -> статистически значимые различия в росте дочерей имеются

In [49]:
x = np.array([172, 177, 158, 170, 178,175, 164, 160, 169, 165])
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 163])

In [50]:
x_mean = np.mean(x)
y_mean = np.mean(y)
x_v = np.var(x, ddof=1)
y_v = np.var(y, ddof=1)
n_x = len(x)
n_y = len(y)
print(f'y_mean={y_mean}')
print(f'x_mean={x_mean}')
print(f'x_v={x_v}')
print(f'y_v={y_v}')
print(f'n_x={n_x}')
print(f'n_y={n_y}')

y_mean=167.5
x_mean=168.8
x_v=48.17777777777778
y_v=50.5
n_x=10
n_y=10


In [51]:
t_emp = (x_mean - y_mean)/np.sqrt(x_v/n_x + y_v/n_y)
t_emp

0.41384114976800535

Возьмём уровень статистической значимоости α=0.05

In [52]:
alpha1 = 0.05
n = n_x
t1 = stats.t.ppf(alpha1/2, df=2*(n-1))
t2 = stats.t.ppf(1-alpha1/2, df=2*(n-1))
t1, t2

(-2.10092204024096, 2.10092204024096)

Так как выборка у нас независимая, то имеем:


In [53]:
stats.ttest_ind(x, y)

Ttest_indResult(statistic=0.4138411497680053, pvalue=0.68387966499013)

- tкрит = 2.1
- t_emp = 0.413
- tкрит>t_emp
- H0>H1. Нулевая гипотеза не отвергается. Статистически значимых различий в росте дочерей нет